Library/Homebrew/formula.rb

```diff
@@ -922,6 +897,41 @@ def plist_startup
     self.class.plist_startup
   end
 
+  # This method can be overridden to provide a plist.
+  # @see https://www.unix.com/man-page/all/5/plist/ Apple's plist(5) man page
+  # <pre>plist do
+  #  {
+  #    Label: plist_name,
+  #    ProgramArguments: ["#{opt_bin}/example", "--do-this"],
+  #    RunAtLoad: true,
+  #    KeepAlive: true,
+  #    StandardErrorPath: "/dev/null",
+  #    StandardOutPath: "/dev/null",
+  #  }
+  # end</pre>
+  def self.plist
```

Comment:
This needs to take a label as argument.

Reply:
However I cannot find a good way to get the formula `name` in here, so we might need to use a placeholder for `plist_name` here and then insert it in `def plist`.

Reply:
> This needs to take a label as argument. And if this argument is `nil`, use `plist_name` instead.

That's basically what the code is already doing as it stands, except this is being passed as the `Label` within the hash instead of as a direct argument to the method, as that seems like a nicer API to me, but this could be switched… are you suggesting that the latter approach will make things run any differently in terms of being able to define multiple `plist do` calls in a single formula?

> However I cannot find a good way to get the formula `name` in here,

This part is the fundamental problem… I'd rather get this working first, before we decide exactly how it looks.

> so we might need to use a placeholder for `plist_name` here and then insert it in `def plist`.

A placeholder? Do you mean some kind of `FORMULA_NAME` string that we `sub` inside of `def plist` just because we have access to `name` there… or something less hacky?

This feels like trying to code inside of a cage 🙈😅

Reply:
> Do you mean some kind of `FORMULA_NAME` string that we `sub` inside of `def plist` just because we have access to `name` there… or something less hacky?

Yes, that's exactly what I mean. The alternative would be to refactor `Formulary` to load the formula class with the name.